astrapy/db.py

```diff
@@ -1753,6 +1761,14 @@ async def delete_subdocument(self, id: str, subdoc: str) -> API_RESPONSE:
         return response
 
     async def upsert(self, document: API_DOC) -> str:
+        # Deprecated: call the upsert_one method
+        DEPRECATION_MESSAGE = "Method 'upsert' of AstraDBCollection is deprecated. Please switch to method 'upsert_one'."
+
+        warn(DEPRECATION_MESSAGE, DeprecationWarning, stacklevel=2)
```

Comment:
The method `logging.warn` is deprecated. `logging.warning` should be used instead.

Reply:
This is `from warnings import warn` being used, probably good as is (no?)

Reply:
Oh, sorry for the noise. It’s logging.warn which is deprecated…